lxd/images.go

```diff
@@ -2637,7 +2637,12 @@ func imageValidSecret(s *state.State, r *http.Request, projectName string, finge
 			continue
 		}
 
-		if !shared.StringInSlice(fmt.Sprintf("/1.0/images/%s", fingerprint), opImages) {
+		opImagesStr := make([]string, len(opImages))
```

Comment:
Pre-filling the slice with empty elements can be quite dangerous as it makes it easy to accidentally introduce bugs in the future if a condition is ever added to the loop below that means that not every element is added to the new slice.

Instead we tend to use the `make([]string,0,len(opImages))` invocation type so that the capacity is still pre-allocated, but not the actual empty elements, we can then use `append` to add the elements to the new slice.